app/src/main/java/org/oppia/android/app/home/recentlyplayed/RecentlyPlayedFragmentPresenter.kt

```diff
@@ -75,67 +75,70 @@ class RecentlyPlayedFragmentPresenter @Inject constructor(
   private fun subscribeToOngoingStoryList() {
     getAssumedSuccessfulPromotedActivityList().observe(
       fragment,
-      Observer<PromotedActivityList> { it ->
-        if (it.promotedStoryList.recentlyPlayedStoryList.isNotEmpty()) {
-          val recentSectionTitleViewModel =
-            SectionTitleViewModel(activity.getString(R.string.ongoing_story_last_week), false)
-          itemList.add(recentSectionTitleViewModel)
-          for (promotedStory in it.promotedStoryList.recentlyPlayedStoryList) {
-            val ongoingStoryViewModel =
-              OngoingStoryViewModel(
-                promotedStory,
-                entityType,
-                fragment as OngoingStoryClickListener
-              )
-            itemList.add(ongoingStoryViewModel)
+      {
+        with(it.promotedStoryList) {
```

Comment:
I think using ``with`` in this way is actually making it harder to understand what's going on.

Plus, this method is starting to get long/hard to understand. Can we break it into smaller helper methods/pieces to make the constituent pieces clearer?

Reply:
I have optimized code PTAL.